numpysocket/numpysocket.py

```diff
@@ -3,45 +3,55 @@
 from io import BytesIO
 import logging
 import socket
-from typing import Any, Union
+from typing import Any
 
 import numpy as np
 
 
 class NumpySocket(socket.socket):
+
+    prev = None
```

Comment:
my major concern with your change is the addition of this state. In the ideal case we let the OS's buffer handle this and just pull bytes from the socket interface.

Adding state to this implementation makes the use of it non-idempotent.